### ISSUE TITLE
[Bug with Fix] Weird Z-axis/Rotation issue in UI when using canvas mode ScreenSpace

### ISSUE THREAD

**Issue by shubhank008** [suriyun-production/mmorpg-kit-docs#1276]
If we switch Canvas Mode to screen space, there is a weird issue in all UI elements of the kit where they appear slightly tilted and have weird Z-axis and rotation alignment.  
In otherwords, most of the UI elements like text, button, etc. become invisible or not visible.  

To fix it, need to edit UIList.cs on line 46~

```
//Fix Z axis bug when using Canvas = Screen Space
                    //ui.transform.localScale = Vector3.one;
                    RectTransform rectTransform = ui.GetComponent<RectTransform>();
                    if(rectTransform != null)
                    {
                        rectTransform.localPosition = Vector3.zero;
                        rectTransform.localRotation = Quaternion.identity;
                        rectTransform.localScale = Vector3.one;
                    }
```

![image](https://user-images.githubusercontent.com/3790163/142295351-40c57447-63a4-4f91-aeba-ae5a2a721ec4.png)

**Comment by insthync** [comment]
It has to be rect transform?, I think just transform will work properly, I will set transform to:

Line 45:
```
                    ui.transform.localPosition = Vector3.zero;
                    ui.transform.localRotation = Quaternion.identity;
                    ui.transform.localScale = Vector3.one;
```

**Comment by shubhank008** [comment]
I actually trial and errored this some time ago and only had to redo it in a new project yesterday so remembered to report it, but cannot remember if using transform works but I think it won't because the problem is in UI elements (buttons/image/text) which uses RectTransform instead.  

But you can try it and make sure :P Change change Canvas mode to Screen Space to see the visual bug

**Comment by insthync** [comment]
Apply the fix in 1.71e
